tests/backup/backup_cluster_share_test.go

```diff
@@ -0,0 +1,481 @@
+				err = CreateBackupLocationWithContext(provider, bkpLocationName, backupLocationUID, cloudCredName, cloudCredUID, getGlobalBucketName(provider), BackupOrgID, "", testUsers[2].ctx, true)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", bkpLocationName))
+			}
+		})
+
+		Step("create a backup and restore it on the same cluster(Shared Cluster) with user(User2)", func() {
+
+			// Take Backup
+			log.InfoD(fmt.Sprintf("Taking backup of multiple namespaces [%v]", bkpNamespaces))
+			err := CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID, bkpNamespaces, nil, BackupOrgID, clusterUid, "", "", "", "", testUsers[2].ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of backup [%s]", backupName))
+
+			// Create Restore
+			for _, namespace := range bkpNamespaces {
+				restoredNameSpace := fmt.Sprintf("%s-%s", RandomString(10), "restored")
+				namespaceMapping[namespace] = restoredNameSpace
+			}
+			log.InfoD("Namespace mapping is %v:", namespaceMapping)
+			err = CreateRestore(restoreName, backupName, namespaceMapping, SourceClusterName, BackupOrgID, testUsers[2].ctx, make(map[string]string))
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of restore [%s]", restoreName))
+			restoreUID, err = Inst().Backup.GetRestoreUID(testUsers[2].ctx, restoreName, BackupOrgID)
+			log.FailOnError(err, "failed to get restore %s uid", restoreName)
+
+		})
+
+		Step("User1 UnShares the cluster from User2", func() {
+			// Un-Share the source cluster with testuser2
+			err := UnShareCluster(SourceClusterName, nil, []string{testUsers[2].name}, testUsers[1].ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying Unshare of source [%s] cluster with %s user using %s ctx", SourceClusterName, testUsers[2].name, testUsers[1].name))
+		})
+
+		Step("Validate the restore object", func() {
+			log.InfoD("validating the restore object")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			restoreInspectRequest := &api.RestoreInspectRequest{
+				Name:  restoreName,
+				OrgId: BackupOrgID,
+				Uid:   restoreUID,
+			}
+			_, err = Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+			log.FailOnNoError(err, "inspect restore %s", restoreName)
+		})
+
+		Step("Cleanup", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			backupDriver := Inst().Backup
+			backupUID, err := backupDriver.GetBackupUID(ctx, backupName, BackupOrgID)
+			log.FailOnError(err, "Failed while trying to get backup UID for - [%s]", backupName)
+			log.InfoD("Deleting backup")
+			_, err = DeleteBackup(backupName, backupUID, BackupOrgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting backup [%s]", backupName))
+		})
+
+	})
+	// This testcase verifies whether the restores created/owned by the user with super-admin role were not deleted during the cluster un-share.
```

Comment:
Can we combine the above test case and this one, both looks similar apart from in the above test case we want to verify restores are deleted in case of normal user and in this we want to verify restores are not deleted in case of super admin. We can have these as two separate steps.

Reply:
As discussed with @vsundarraj-px he mentioned it is better to maintain the isolation.